domain/src/main/java/org/oppia/domain/topic/TopicController.kt

```diff
@@ -821,16 +699,24 @@ class TopicController @Inject constructor(
             )
           ).build()
       )
-      .addAllWorkedExample(
-        createWorkedExamplesFromJson(
-          skillContents.getJSONArray(
-            "worked_examples"
-          )
-        )
-      )
+      .addAllWorkedExample(workedExamplesList)
+      .putAllWrittenTranslation(writtenTranslationMapping)
+      .putAllRecordedVoiceover(recordedVoiceoverMapping)
       .build()
   }
 
+  private fun getSkillJsonObject(skillId: String): JSONObject {
+    val skillJsonArray = jsonAssetRetriever.loadJsonFromAsset("skills.json")?.optJSONArray("skills")
+      ?: return JSONObject("")
+    for (i in 0 until skillJsonArray.length()) {
+      val currentSkillJsonObject = skillJsonArray.optJSONObject(i)
+      if (skillId == currentSkillJsonObject.optString("id")) {
+        return currentSkillJsonObject
+      }
+    }
+    return return JSONObject("")
```

Comment:
nit change, one return, else this will be unreachable code.

Reply:
Took me a minute to realise this mistake. :-D Done.

Reply:
Thanks :)